cobra/ui/model.go

```diff
@@ -0,0 +1,365 @@
+/*
+ *  MIT License
+ *
+ *  Copyright (c) 2020 Nicolas JUHEL
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in all
+ *  copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ *  SOFTWARE.
+ *
+ */
+
+package ui
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	tea "github.com/charmbracelet/bubbletea"
+	spfcbr "github.com/spf13/cobra"
+)
+
+const pageSize = 10
+
+type ui struct {
+	cobra       *spfcbr.Command
+	questions   []Question
+	index       int
+	input       string
+	cursor      int
+	errorMsg    string
+	userAnswers []string
+	LastMessage string
+}
+
+func (u *ui) SetCobra(cobra *spfcbr.Command) {
+	u.cobra = cobra
+}
+
+func (u *ui) Init() tea.Cmd {
+	return nil
+}
+
+func (u *ui) SetLastMessage(msg string) {
+	u.LastMessage = msg
+}
+
+func (u *ui) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
+	var err error
+	if u.index >= len(u.questions) {
+		return u, tea.Quit
+	}
+	switch msg := msg.(type) {
+	case tea.KeyMsg:
+		if msg.String() == "ctrl+c" {
+			return u, tea.Quit
+		}
+		switch msg.Type {
+		case tea.KeyEnter:
+			if u.index < len(u.questions) {
+				if u.questions[u.index].FilePath {
```

Comment:
make a sub private function to manage filepath statement like 
`
func (u *ui) questionFilePath() {
...
}
` 
and replace this inline statement to call of this sub function like this: 
`if u.questions[u.index].FilePath {
    u.questionFilePath()
}`

Reply:
Done